### DISCUSSION TITLE
How to disable weekends from the date range picker?

### DISCUSSION THREAD

**Discussion by Shaker-Hamdi** [rsuite/rsuite#3089] [Q&A] [answered]
Hello,

I need to disable the weekends (Friday and Saturday) from the date range selection, how can I do that?
I need to be able to select a date range that includes only weekdays and ignore the weekend days.
Right now I'm able to disable "before today" and "after today", but I can't find how to disable weekends.

**Comment by SevenOutman** [comment, marked as answer]
Hi @Shaker-Hamdi , focusing on this specific case, I don't think DateRangePicker *alone* is suitable for usage because

- Its UI pattern can't express "from this day to that day, *but except for these days*"
- Its data structure can't express that either - `value` is a Date tuple containing only starting and ending date rather than an array of all covered dates

I would suggest use DateRangePicker only for selecting the beginning and ending of the date range, then use another widget for telling users that weekends are excluded.

**Reply by Shaker-Hamdi** [reply]
Thank you @SevenOutman. I ended up doing exactly what you're suggesting. Thank you.
